pkg/apis/core/v1alpha1/objectselector.go

```diff
@@ -0,0 +1,21 @@
+package v1alpha1
+
+// Selector for any Kubernetes-style API.
+type ObjectSelector struct {
```

Comment:
Not critical now because it happens later anyway, but it might be worth making this implement `Validator` and ensure the regexes compile so you get an error upfront when trying to create/edit the entity. (I'm 80% sure that the codegen recursively calls validate functions if subfields implement `Validator`)